app/src/sharedTest/java/org/oppia/android/app/parser/CustomBulletSpanTest.kt

```diff
@@ -0,0 +1,238 @@
+import org.oppia.android.domain.topic.PrimeTopicAssetsControllerModule
+import org.oppia.android.testing.TestAccessibilityModule
+import org.oppia.android.testing.TestDispatcherModule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.CustomBulletSpan
+import org.oppia.android.util.parser.GlideImageLoaderModule
+import org.oppia.android.util.parser.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Singleton
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = CustomBulletSpanTest.TestApplication::class, qualifiers = "port-xxhdpi")
+class CustomBulletSpanTest {
+
+  private var context: Context = ApplicationProvider.getApplicationContext<TestApplication>()
+
+  val testStringWithoutBulletSpan = SpannableStringBuilder("Text Without BulletSpan")
+  val testStringWithBulletSpan = SpannableStringBuilder("Text With \nBullet Point").apply {
+    setSpan(BulletSpan(), 10, 22, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
+  }
+  val testStringWithMultipleBulletSpan = SpannableStringBuilder(
+    "Text With \nfirst \nsecond \nthird \nfour \nfive"
+  ).apply {
+    setSpan(BulletSpan(), 10, 18, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
+    setSpan(BulletSpan(), 18, 27, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
+    setSpan(BulletSpan(), 27, 35, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
+    setSpan(BulletSpan(), 35, 42, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
+    setSpan(UnderlineSpan(), 42, 43, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
+  }
+  val testStringWithCustomBulletSpan = SpannableStringBuilder("Text With \nBullet Point").apply {
+    this.setSpan(
+      CustomBulletSpan(context),
+      10,
+      22,
+      Spannable.SPAN_EXCLUSIVE_EXCLUSIVE
+    )
+  }
+
+  @Test
+  fun customBulletSpan_testReplaceBulletSpan_spannableStringWithoutBulletSpanRemainSame() {
+    val spannableString1 = testStringWithoutBulletSpan
+    val spannableString2 = CustomBulletSpan.replaceBulletSpan(spannableString1, context)
+    assertThat(getNoofBulletSpans(spannableString1)).isEqualTo(0)
+    assertThat(getNoofCustomBulletSpans(spannableString1)).isEqualTo(0)
+    assertThat(getNoofBulletSpans(spannableString2)).isEqualTo(0)
+    assertThat(getNoofCustomBulletSpans(spannableString2)).isEqualTo(0)
+  }
+
+  @Test
+  fun customBulletSpan_testReplaceBulletSpan_spannableStringWithBulletSpanIsNotSame() {
+    val spannableString1 = testStringWithBulletSpan
+    val spannableString2 = CustomBulletSpan.replaceBulletSpan(spannableString1, context)
+//    assertThat(getNoofBulletSpans(spannableString1)).isEqualTo(1) //0
```

Comment:
For a string with one bulletSpan but it gives 0 no. of BulletSpans when I run this testcase

Reply:
Ah, I think you might be running into a shared state issue. Your test values are SpannableStringBuilders which means they're mutable, and replaceBulletSpan is actually changing the builder state. You should be storing SpannableStrings as the test variables & converting them to/from builders only when doing the replace step. I think that will address your issue.